hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -140,8 +140,17 @@
 
     private final HazelcastThreadGroup hazelcastThreadGroup;
 
+
     private final boolean liteMember;
 
+    protected NodeEngineImpl createNodeEngine() {
```

Comment:
Can you add some comments that this method needs to be protected for outside projects? Else someone could make it private